### DISCUSSION TITLE
In opentcs-Integration-Example, I am able to connect with Broker, but Vehicle is not being Connected.

### DISCUSSION THREAD

**Discussion by Gosling1702** [openTCS/opentcs#151] [General]
I am working on the openTCS-integration-Example and I tried to integrated it with VDA5050 adapter v2.0. After Kernel, Model Editor, Operational Desk, Kernel Control Center service turned UP. I am getting in openTCS template VDA Adapter for Operational Desk as-

While I enable the VDA5050 adapter, the Vehicle state is still "Unknown". and My Control Center is-

### It will connected with broker, but not with the Vehicle. how can I enable Vehicle connection.

**Comment by Sadhu17** [comment]
Hello Peter, [z29591259](https://github.com/z29591259), [martingr](https://github.com/martingr), [swltr](https://github.com/swltr)

I am integrating the VDA5050 version 2.0 adapter with Opentcs-5.11.0 and encountering some issues. The Vehicle and Broker connect successfully at the KernelControlCenter, but problems arise when I try to move the vehicle at the OperationalDesk. I would appreciate any guidance on how to resolve the following issues:

1.- After enabling the vehicle and changing its integration level to "Integrated Fully", the vehicle moves erratically and in reverse.
2.- The vehicle’s movement path is not updating correctly, and previously traversed paths are not being cleared.
3.- Once the vehicle reaches its destination, the order status remains "Being Processed," preventing the vehicle from processing the next queued order.
4.- Is there an alternative method for assigning multiple orders to the vehicle?
5.- The message containing "orderId": "TOrder-01J1YQG11PPD8FN7HWNCTZC6ZH-0" and "orderUpdateId": 1 is being published continuously having same OrderId and orderUpdateId like {orderId": "TOrder-01J1YQG11PPD8FN7HWNCTZC6ZH-0" and "orderUpdateId"}, but ideally it has to be published once?

I am also including my simulation topic and message for reference.

Topic: uagv/v2/XYZ/0001/order

Message{
"headerId" : 162,
"timestamp" : "2024-07-04T11:24:24.247443300Z",
"version" : "2.0.0",
"manufacturer" : "XYZ",
"serialNumber" : "0001",
"orderId" : "TOrder-01J1YQG11PPD8FN7HWNCTZC6ZH-0",
"orderUpdateId" : 1,
"nodes" : [ {
"nodeId" : "P-0061",
"sequenceId" : 2,
"released" : true,
"actions" : [ ],
"nodePosition" : {
"x" : -17.25,
"y" : -15.74,
"mapId" : ""
}
}, {
"nodeId" : "P-0009",
"sequenceId" : 4,
"released" : true,
"actions" : [ ],
"nodePosition" : {
"x" : -17.25,
"y" : -13.448,
"mapId" : ""
}
}, {
"nodeId" : "P-0063",
"sequenceId" : 6,
"released" : false,
"actions" : [ ],
"nodePosition" : {
"x" : -17.25,
"y" : -10.81,
"mapId" : ""
}
}, {
"nodeId" : "P-0067",
"sequenceId" : 8,
"released" : false,
"actions" : [ ],
"nodePosition" : {
"x" : -15.5,
"y" : -10.81,
"mapId" : ""
}
}, {
"nodeId" : "P-0071",
"sequenceId" : 10,
"released" : false,
"actions" : [ ],
"nodePosition" : {
"x" : -13.64,
"y" : -10.81,
"mapId" : ""
}
}, {
"nodeId" : "P-0075",
"sequenceId" : 12,
"released" : false,
"actions" : [ ],
"nodePosition" : {
"x" : -11.82,
"y" : -10.81,
"mapId" : ""
}
}, {
"nodeId" : "P-0079",
"sequenceId" : 14,
"released" : false,
"actions" : [ ],
"nodePosition" : {
"x" : -10.05,
"y" : -10.81,
"mapId" : ""
}
}, {
"nodeId" : "P-0087",
"sequenceId" : 16,
"released" : false,
"actions" : [ ],
"nodePosition" : {
"x" : -8.36,
"y" : -10.81,
"mapId" : ""
}
}, {
"nodeId" : "P-0006",
"sequenceId" : 18,
"released" : false,
"actions" : [ ],
"nodePosition" : {
"x" : -8.36,
"y" : -7.0,
"mapId" : ""
}
}, {
"nodeId" : "P-0023",
"sequenceId" : 20,
"released" : false,
"actions" : [ ],
"nodePosition" : {
"x" : -8.36,
"y" : -1.52,
"mapId" : ""
}
}, {
"nodeId" : "P-0022",
"sequenceId" : 22,
"released" : false,
"actions" : [ ],
"nodePosition" : {
"x" : -12.68,
"y" : -1.52,
"mapId" : ""
}
}, {
"nodeId" : "P-0038",
"sequenceId" : 24,
"released" : false,
"actions" : [ ],
"nodePosition" : {
"x" : -17.25,
"y" : -1.5,
"mapId" : ""
}
}, {
"nodeId" : "P-0021",
"sequenceId" : 26,
"released" : false,
"actions" : [ ],
"nodePosition" : {
"x" : -21.33,
"y" : -1.52,
"mapId" : ""
}
}, {
"nodeId" : "P-0007",
"sequenceId" : 28,
"released" : false,
"actions" : [ ],
"nodePosition" : {
"x" : -25.6,
"y" : -1.5,
"mapId" : ""
}
}, {
"nodeId" : "P-0008",
"sequenceId" : 30,
"released" : false,
"actions" : [ ],
"nodePosition" : {
"x" : -25.6,
"y" : 2.5,
"mapId" : ""
}
}, {
"nodeId" : "P-0025",
"sequenceId" : 32,
"released" : false,
"actions" : [ ],
"nodePosition" : {
"x" : -21.33,
"y" : 2.51,
"mapId" : ""
}
}, {
"nodeId" : "P-0029",
"sequenceId" : 34,
"released" : false,
"actions" : [ ],
"nodePosition" : {
"x" : -17.54,
"y" : 2.53,
"mapId" : ""
}
}, {
"nodeId" : "P-0037",
"sequenceId" : 36,
"released" : false,
"actions" : [ ],
"nodePosition" : {
"x" : -17.54,
"y" : 5.3,
"mapId" : ""
}
}, {
"nodeId" : "P-0039",
"sequenceId" : 38,
"released" : false,
"actions" : [ ],
"nodePosition" : {
"x" : -14.47,
"y" : 5.3,
"mapId" : ""
}
}, {
"nodeId" : "P-0040",
"sequenceId" : 40,
"released" : false,
"actions" : [ ],
"nodePosition" : {
"x" : -14.47,
"y" : 8.09,
"mapId" : ""
}
} ],
"edges" : [ {
"edgeId" : "P-0061 --- P-0009",
"sequenceId" : 3,
"released" : true,
"startNodeId" : "P-0061",
"endNodeId" : "P-0009",
"actions" : [ ],
"maxSpeed" : 1.0
}, {
"edgeId" : "P-0009 --- P-0063",
"sequenceId" : 5,
"released" : false,
"startNodeId" : "P-0009",
"endNodeId" : "P-0063",
"actions" : [ ],
"maxSpeed" : 1.0
}, {
"edgeId" : "P-0063 --- P-0067",
"sequenceId" : 7,
"released" : false,
"startNodeId" : "P-0063",
"endNodeId" : "P-0067",
"actions" : [ ],
"maxSpeed" : 1.0
}, {
"edgeId" : "P-0067 --- P-0071",
"sequenceId" : 9,
"released" : false,
"startNodeId" : "P-0067",
"endNodeId" : "P-0071",
"actions" : [ ],
"maxSpeed" : 1.0
}, {
"edgeId" : "P-0071 --- P-0075",
"sequenceId" : 11,
"released" : false,
"startNodeId" : "P-0071",
"endNodeId" : "P-0075",
"actions" : [ ],
"maxSpeed" : 1.0
}, {
"edgeId" : "P-0075 --- P-0079",
"sequenceId" : 13,
"released" : false,
"startNodeId" : "P-0075",
"endNodeId" : "P-0079",
"actions" : [ ],
"maxSpeed" : 1.0
}, {
"edgeId" : "P-0079 --- P-0087",
"sequenceId" : 15,
"released" : false,
"startNodeId" : "P-0079",
"endNodeId" : "P-0087",
"actions" : [ ],
"maxSpeed" : 1.0
}, {
"edgeId" : "P-0087 --- P-0006",
"sequenceId" : 17,
"released" : false,
"startNodeId" : "P-0087",
"endNodeId" : "P-0006",
"actions" : [ ],
"maxSpeed" : 1.0
}, {
"edgeId" : "P-0006 --- P-0023",
"sequenceId" : 19,
"released" : false,
"startNodeId" : "P-0006",
"endNodeId" : "P-0023",
"actions" : [ ],
"maxSpeed" : 1.0
}, {
"edgeId" : "P-0023 --- P-0022",
"sequenceId" : 21,
"released" : false,
"startNodeId" : "P-0023",
"endNodeId" : "P-0022",
"actions" : [ ],
"maxSpeed" : 1.0
}, {
"edgeId" : "P-0022 --- P-0038",
"sequenceId" : 23,
"released" : false,
"startNodeId" : "P-0022",
"endNodeId" : "P-0038",
"actions" : [ ],
"maxSpeed" : 1.0
}, {
"edgeId" : "P-0038 --- P-0021",
"sequenceId" : 25,
"released" : false,
"startNodeId" : "P-0038",
"endNodeId" : "P-0021",
"actions" : [ ],
"maxSpeed" : 1.0
}, {
"edgeId" : "P-0021 --- P-0007",
"sequenceId" : 27,
"released" : false,
"startNodeId" : "P-0021",
"endNodeId" : "P-0007",
"actions" : [ ],
"maxSpeed" : 1.0
}, {
"edgeId" : "P-0007 --- P-0008",
"sequenceId" : 29,
"released" : false,
"startNodeId" : "P-0007",
"endNodeId" : "P-0008",
"actions" : [ ],
"maxSpeed" : 1.0
}, {
"edgeId" : "P-0008 --- P-0025",
"sequenceId" : 31,
"released" : false,
"startNodeId" : "P-0008",
"endNodeId" : "P-0025",
"actions" : [ ],
"maxSpeed" : 1.0
}, {
"edgeId" : "P-0025 --- P-0029",
"sequenceId" : 33,
"released" : false,
"startNodeId" : "P-0025",
"endNodeId" : "P-0029",
"actions" : [ ],
"maxSpeed" : 1.0
}, {
"edgeId" : "P-0029 --- P-0037",
"sequenceId" : 35,
"released" : false,
"startNodeId" : "P-0029",
"endNodeId" : "P-0037",
"actions" : [ ],
"maxSpeed" : 1.0
}, {
"edgeId" : "P-0037 --- P-0039",
"sequenceId" : 37,
"released" : false,
"startNodeId" : "P-0037",
"endNodeId" : "P-0039",
"actions" : [ ],
"maxSpeed" : 1.0
}, {
"edgeId" : "P-0039 --- P-0040",
"sequenceId" : 39,
"released" : false,
"startNodeId" : "P-0039",
"endNodeId" : "P-0040",
"actions" : [ ],
"maxSpeed" : 1.0
},

**Reply by chengqi-wjs** [reply]
Hello, the Vehicle how to connect with the KernelControlCenter

**Comment by chengqi-wjs** [comment]
Did you solve this problem

**Comment by freekevinCode** [comment]
Vehicle not connected mqtt broker is not openTCS problem！

**Comment by chengqi-wjs** [comment]
How can I match the connected vehicle into the map?
